CONTRIBUTING.md

```diff
@@ -310,42 +310,97 @@ We recommend you use IntelliJ as your IDE. The code style template for the proje
-            * How does it address the issue?
-            * What effects does the patch have?
-            * Do not assume the reviewer understands what the original problem was.
-            * Do not assume the code is self-evident or self-documenting.
-            * Read the commit message to see if it hints at improved code structure.
-            * The first commit line is the most important.
-            * Describe any limitations of the current code.
-            * Do not include patch set-specific comments.
-
-Details for each point and good commit message examples can be found on https://wiki.openstack.org/wiki/GitCommitMessages#Information_in_commit_messages
+
+### Conventional Commits
+We follow the [Conventional Commits](https://www.conventionalcommits.org/) specification for our commit messages and PR titles.
+
+**PR Title Format:**
+```
+<type>[(scope)]: <description>
+```
+
+**Types:**
+* **feat**: New feature or functionality
+* **fix**: Bug fix
+* **docs**: Documentation only changes
+* **style**: Code style changes (formatting, semicolons, etc)
+* **refactor**: Code refactoring without changing functionality
+* **perf**: Performance improvements
+* **test**: Adding or modifying tests
+* **build**: Build system or dependency changes
+* **ci**: CI/CD configuration changes
+* **chore**: Maintenance tasks
+* **revert**: Reverting a previous commit
+
+Note: Each PR/commit should have a single primary type. If your changes span multiple categories, choose the most significant one or consider splitting into separate PRs.
+
+**Scope (optional):** The area of code affected. Common scopes include:
+* `parser` - SQL parser and grammar
+* `analyzer` - Query analysis and validation
+* `planner` - Query planning, optimization, and rules (including CBO)
+* `spi` - Service Provider Interface changes
+* `scheduler` - Task scheduling and execution
+* `protocol` - Wire protocol and serialization
+* `connector` - Changes to connector functionality
+* `resource` - Resource management (memory manager, resource groups)
+* `security` - Authentication and authorization
+* `function` - Built-in functions and operators
+* `type` - Type system and type coercion
+* `expression` - Expression evaluation
+* `operator` - Query operators (join, aggregation, etc.)
+* `client` - Client libraries and protocols
+* `server` - Server configuration and management
+* `testing` - Test framework and utilities
+* `docs` - Documentation
+* `build` - Build system and dependencies
+
+An acceptable scope may also be any connector included in this repository, such as `hive` and `iceberg`.
+
+**Description:**
+* Must start with a capital letter
+* Must not end with a period
+* Use imperative mood ("Add feature" not "Added feature")
+* Be concise but descriptive
+
+**Breaking Changes:**
+* Use `!` after the type/scope (e.g., `feat!: Remove deprecated API`)
+* AND include `BREAKING CHANGE:` in the PR description footer with a detailed explanation of the change
```

Comment:
To confirm, did you intentionally leave it as PR description footer, as we are mandating use of  `!` in the commit title, or should this be changed to commit description footer?

Reply:
Unintentional, updated